### PR TITLE
Cadence HiFi4 NN Library v2.2.0 update

### PR DESCRIPTION
**Pull request by pnikam-cad**
The following changes are done to the HiFi4 implementation of TF Lite Micro

1. Update the xtensa_hifi kernel files and the make setup to use HiFi4 NN Library v2.2.0 .
2. Update the kernel files as per the latest reference implementation.
3. Add xtenas_hifi kernel implementations for the add and mul operators.

### CODE REVIEW

**Review by njeffrie — DISMISSED** [review]
I have a few style nits. Other than that, looks good.

File: tensorflow/lite/micro/kernels/xtensa_hifi/activations.cc [hunk 1]
```diff
@@ -140,14 +143,15 @@ TfLiteStatus ReluEval(TfLiteContext* context, TfLiteNode* node) {
...
       const int flat_size = MatchingFlatSize(input_shape, output_shape);
+      const uint8_t zero = input->params.zero_point;
 
       inp_data_ptr = GetTensorData<uint8_t>(input);
       out_data_ptr = GetTensorData<uint8_t>(output);
 
-      err = xa_nn_vec_activation_min_max_asym8_asym8(
-          out_data_ptr, inp_data_ptr, 0, 255, flat_size);  // Is 255 right?
+      err = xa_nn_vec_activation_min_max_asym8_asym8(out_data_ptr, inp_data_ptr,
+                                                     zero, 255, flat_size);
```
[the remark this answers is not in the document]
Comment (pnikam-cad) [reply tensorflow/lite/micro/kernels/xtensa_hifi/activations.cc]: Updated as per suggestion.

File: tensorflow/lite/micro/kernels/xtensa_hifi/add.cc [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+TfLiteStatus EvalAdd(TfLiteContext* context, TfLiteNode* node, TfLiteAddParams* params,
+             const OpData* data, const TfLiteTensor* input1,
+             const TfLiteTensor* input2, TfLiteTensor* output) {
+  float output_activation_min, output_activation_max;
+  CalculateActivationRange(params->activation, &output_activation_min,
+                           &output_activation_max);
+  tflite::ArithmeticParams op_params;
+  SetActivationParams(output_activation_min, output_activation_max, &op_params);
+#define TF_LITE_ADD(opname)                                                   \
```
[the remark this answers is not in the document]
Comment (pnikam-cad) [reply tensorflow/lite/micro/kernels/xtensa_hifi/add.cc:107]: Here we need the macro at couple of places, so keeping it. When the vector floating point unit (VFPU) hardware is not available, we need to fall back to the reference implementation.

File: tensorflow/lite/micro/kernels/xtensa_hifi/conv.cc [hunk 3]
```diff
@@ -186,63 +242,53 @@ TfLiteStatus EvalQuantized(TfLiteContext* context, TfLiteNode* node,
...
     if (required_scratch <= 0) {
       TF_LITE_KERNEL_LOG(context,
-                         "conv2d_std_asym8: xa_nn_conv2d_std_getsize failed");
+          "conv2d_std_asym8: xa_nn_conv2d_std_getsize failed");
       return kTfLiteError;
     }
 
     ALLOCATE_XTENSA_NNLIB_SCRATCH_MEM;
-    p_scratch = xtensa_nnlib_scratch_buf;
+    p_scratch = (char*)xtensa_nnlib_scratch_buf;
```
[the remark this answers is not in the document]
Comment (pnikam-cad) [reply tensorflow/lite/micro/kernels/xtensa_hifi/conv.cc]: Updated the code in conv.cc and depthwise_conv.cc to clean up type casts.

**Comment by pnikam-cad** [comment]
I have updated the PR to address the above suggestions, please review.

File: tensorflow/lite/micro/kernels/xtensa_hifi/add.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (njeffrie) [reply tensorflow/lite/micro/kernels/xtensa_hifi/add.cc:107]: Looks like I missed some of the uses below.  Looks good.

**Review by njeffrie — APPROVED** [review]

**Pull request merged**